src/Forms/FocusPointField.php

```diff
@@ -42,21 +40,29 @@ class FocusPointField extends FieldGroup
      */
     private static $max_height = 150;
 
-    public function __construct(Image $image)
+    public function __construct($name, $title = null, Image $image = null)
     {
         // Create the fields
-        $previewImage = $image->FitMax($this->config()->get('max_width'), $this->config()->get('max_height'));
-        $fields = array(
-            LiteralField::create('FocusPointGrid', $previewImage->renderWith(FocusPointField::class)),
-            TextField::create('FocusX'),
-            TextField::create('FocusY'),
-        );
+        $fields = [
+            TextField::create($name . 'FocusX'),
+            TextField::create($name . 'FocusY')
+        ];
+
+        if ($image) {
+            $previewImage = $image->FitMax($this->config()->get('max_width'), $this->config()->get('max_height'));
+            array_unshift($fields, LiteralField::create('FocusPointGrid', $previewImage->renderWith(
+                FocusPointField::class,
+                ['FieldGridBackgroundCSS' => $image->FieldGridBackgroundCSS($previewImage->getWidth(), $previewImage->getHeight())]
```

Comment:
Somehow the template was rendered with a `DBFile` context and didn't have the focuspoint set… I imagine the focuspoint was lost during `FitMax`.
Not sure what's a good way to circumvent that. Overall, the template rendering in this instance is moot, as it'll be superseded by the react component.

Reply:
Ah, I know what the issue was. Somehow the `FocusPointImageExtension` was also applied to `DBFile`. That's not really a good idea, as `DBFile` should already have access to the image props/methods via failover. I'll investigate further.

Reply:
`DBFile` uses the `ImageManipulation` trait, so presumably it needs to receive most (not all) of the properties and methods in `FocusPointImageExtension`. I'm guessing that resampled image objects are `DBFiles` rather than `Images` in some contexts but I got lost trying to understand that.

Reply:
`ImageManipulation` always sets the original file on the `DBFile` after a manipulation. So in `ImageManipulation::manipulate`, when it created a new variant of an `Image`, it'll return a `DBFile` instance with the original Image as failover.

So any method you call on `DBFile` that doesn't exist there will call the method on the original. Therefore an additional field or method should be inherited automatically.

Reply:
> So any method you call on DBFile that doesn't exist there will call the method on the original. Therefore any additional field or method should be inherited automatically.

Interesting. Bear with me, this concept is new to me. So it would call the method on the original Image object? That would interfere with chaining then right? Like you couldn't go `$Image.Greyscale.FocusFill(50,50)` for example because you would just end up with `$Image.FocusFill(50,50)`?

I also want to pass through updated FocusPoint coords after a crop, adjusted to represent the same point in the cropped image. This is important for me in case you want to do anything with the focus point in the front end, for instance set the css transform point to the same location, or integrate with scripts like jQuery FocusPoint.